scripts/build-api.js

```diff
@@ -1,40 +1,36 @@
 #!/usr/bin/env node
-
-import { readFile, writeFile, unlink, mkdir } from 'fs/promises'
-import { fileURLToPath } from 'url'
+import {  writeFile, mkdir, rm } from 'fs/promises'
 import remarkHighlight from 'remark-highlight.js'
 import { existsSync } from 'fs'
 import { promisify } from 'util'
 import childProcess from 'child_process'
 import remarkParse from 'remark-parse'
 import { unified } from 'unified'
 import remarkHtml from 'remark-html'
-import parcelCore from '@parcel/core'
 import { globby } from 'globby'
 import { join } from 'path'
 import TypeDoc from 'typedoc'
+import vite from 'vite'
+import vitePugPlugin from 'vite-plugin-pug-transformer'
 
-let exec = promisify(childProcess.exec)
-let Parcel = parcelCore.default
+import { PROJECTS, DIST, SRC } from './lib/dir.js'
 
-const ROOT = join(fileURLToPath(import.meta.url), '..', '..')
-const PROJECTS = join(ROOT, '..')
-const DIST = join(ROOT, 'dist')
-const SRC = join(ROOT, 'src')
+let exec = promisify(childProcess.exec)
 
 async function buildLayout() {
-  let bundler = new Parcel({
-    entries: join(SRC, 'api.pug'),
-    defaultConfig: join(ROOT, 'node_modules', '@parcel', 'config-default'),
-    patchConsole: false,
-    sourceMaps: false,
-    mode: 'production'
+  let data = await vite.build({
+    plugins: [vitePugPlugin()],
```

Comment:
We can try to use [conditional vite.config](https://vitejs.dev/config/#conditional-config). We can pass argument to build commands (or use env variable) to specify what type of build starts and change settings like `outDir` in vite.config. This will remove duplication and leave the configuration in one place

@ai what do you think?